test/distributed/org/apache/cassandra/distributed/test/JMXSslConfigDistributedTest.java

```diff
@@ -0,0 +1,204 @@
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_CIPHER_SUITES.reset();
+        JAVAX_RMI_SSL_CLIENT_ENABLED_PROTOCOLS.reset();
+        JAVAX_RMI_SSL_CLIENT_ENABLED_CIPHER_SUITES.reset();
+    }
+
+    @SuppressWarnings("unchecked")
+    private void configureClientSocketFactory(Map<String, Object> jmxEnv, Map<String, Object> encryptionOptionsMap) throws SSLException
+    {
+        JmxTestClientSslContextFactory clientSslContextFactory = new JmxTestClientSslContextFactory(encryptionOptionsMap);
+        List<String> cipherSuitesList = (List<String>) encryptionOptionsMap.get("cipher_suites");
+        String[] cipherSuites = cipherSuitesList == null ? null : cipherSuitesList.toArray(new String[0]);
+        List<String> acceptedProtocolList = (List<String>) encryptionOptionsMap.get("accepted_protocols");
+        String[] acceptedProtocols = acceptedProtocolList == null ? null : acceptedProtocolList.toArray(new String[0]);
+        JmxTestClientSslSocketFactory clientFactory = new JmxTestClientSslSocketFactory(clientSslContextFactory.createSSLContext(),
+                                                                                        cipherSuites, acceptedProtocols);
+        jmxEnv.put(RMIConnectorServer.RMI_CLIENT_SOCKET_FACTORY_ATTRIBUTE, clientFactory);
+        jmxEnv.put("com.sun.jndi.rmi.factory.socket", clientFactory);
+    }
+
+    @Test
+    public void testDefaultEncryptionOptions() throws Throwable
+    {
+        setSystemTrustStore((String)validKeystore.get("truststore"), (String)validKeystore.get("truststore_password"));
+        // We must set the keystore in the system variable to make sure that the call to SSLContext.getDefault()
+        // uses it when Client SSL Socketfactory is initialized even if we don't need it here.
+        // The same default SSLContext.getDefault() will be used by other methods like testSystemSettings() in this test
+        // for the Server SSL Socketfactory and at that time we will need the keystore to be available
+        // All of the above is the issue because we run everything (JMX Server, Client) in the same JVM, multiple times
+        // and the SSLContext.getDefault() relies on static initialization that is reused
+        setSystemKeyStore((String)validKeystore.get("keystore"), (String)validKeystore.get("keystore_password"));
+        ImmutableMap<String, Object> encryptionOptionsMap = ImmutableMap.<String, Object>builder().putAll(validKeystore)
+                                                                        .put("enabled", true)
+                                                                        .put("accepted_protocols", Arrays.asList("TLSv1.2", "TLSv1.3", "TLSv1.1"))
+                                                                        .build();
+
+        try (Cluster cluster = builder().withNodes(1).withConfig(c -> {
+            c.with(Feature.JMX);
+            c.set("jmx_encryption_options", encryptionOptionsMap);
```

Comment:
just inline options directly here, if you use WithProperties too, you'll have it pretty nice.

Reply:
I didn't quite get your suggestion. Can you please suggest the change once I update the PR with other comments resolutions?

Reply:
Okay, I updated the PR with addressing other comments. Now, you can make a code change suggestion.